### ISSUE TITLE
[HOLD for payment 2024-08-23] [$500] Some chats are showing as bold (unread) in the chat search, but showing as non-bold (read) in the LHN

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#46423]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** Needs Reproduction
**Reproducible in production?:** Needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @tgolen 
**Slack conversation:** https://expensify.slack.com/archives/C05LX9D6E07/p1722266268926929

## Action Performed:

1. Open up the chat search
2. Look for some group channels like #social

## Expected Result:
The same chats that are showing bold in the LHN should be shown as bold in the chat search results

## Actual Result:
Sometimes, chats will be bolded in the search results when they are not bolded in the LHN. This can happen for chats that you have muted (the search results will still show them as unread, while the LHN shows them as read).

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [X] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![image](https://github.com/user-attachments/assets/c1eca085-779d-4d8c-bf2b-c282d548a752)
![image](https://github.com/user-attachments/assets/ce18a09e-7d65-483c-b5bf-80a50e8b1f60)

[log.txt](https://github.com/user-attachments/files/16416088/log.txt)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01803b7fcbf92fa7d7</li>
        <li>Upwork Job ID: 1818654097937596043</li>
        <li>Last Price Increase: 2024-07-31</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @s77rt</details>

**Comment by iwiznia** [comment]
We might be just removing the bolding there. Discussion here https://expensify.slack.com/archives/C03U7DCU4/p1722269953413229

**Comment by tgolen** [comment]
In that thread, we had 8 👍 for removing the bold styling from the search options, so let's pursue that.

**Comment by s77rt** [comment]
@daledah Thanks for the proposal. RCA and solution looks good to me.

🎀 👀 🎀 C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/46423#issuecomment-2261211424)

**Comment by tgolen** [comment]
Let me ask about that! I'll get back to you on it.

**Comment by tgolen** [comment]
Sorry... I'm going to put a HOLD on this for a second. We had some more discussion internally, and we'd like to change the assumption of this GH.

We'd actually like to better understand why the search results do not match the LHN. In other words, why does the LHN show the chat as unread, but the search results show it as being read? There must be some kind of a bug or issue happening in there. Rather than just covering it up by never showing them as bold, let's instead focus on fixing the issue with the read/unread status.

@daledah Is this something you can look into and can reproduce or do you need more debugging or troubleshooting from me?

**Comment by tgolen** [comment]
Wow, thanks. OK, I realized today that I had the `#social` channel muted 😊 but... I still think the search results should mimic what the LHN does in both cases above. Would you be willing to include those changes on your PR? If it increases the scope enough, I can consider bumping up the price of the issue to account for it.

**Comment by tgolen** [comment]
Yeah, as I mentioned in my previous comment:

> We had some more discussion internally, and we'd like to change the assumption of this GH.

I will adjust the description and title of the GH.

**Comment by tgolen** [comment]
Oh, thanks for the reminder. I've removed the HOLD.

**Comment by abekkala** [comment]
@daledah is there an update on the PR for this one?

**Comment by s77rt** [comment]
@abekkala [PR](https://github.com/Expensify/App/pull/46645) is merged and deployed to production 5 days ago

**Comment by abekkala** [comment]
Deployed to prod. August 16

### PAYMENT SUMMARY FOR AUG 23, if no regressions

- Fix: @daledah [$250] [Offer](https://www.upwork.com/nx/wm/offer/103619863)
- PR Review: @s77rt [$250]  Payment via NewDot
Please complete checklist

______

**BugZero Checklist**: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ]  [@s77rt] The PR that introduced the bug has been identified. Link to the PR:
- [ ]  [@s77rt] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@s77rt] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ]  [@s77rt] Determine if we should create a regression test for this bug.
- [ ]  [@s77rt] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ]  [@abekkala] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: https://github.com/Expensify/App/pull/37306
- The offending PR has been commented on: https://github.com/Expensify/App/pull/37306#discussion_r1725451224
- A discussion in #expensify-bugs has been started: Not needed
- Determine if we should create a regression test for this bug: No

**Comment by tgolen** [comment]
Yeah, I think increasing the price is appropriate. @abekkala Could you please increase the price of this issue to $500?

**Comment by abekkala** [comment]
### UPDATED PAYMENT SUMMARY FOR AUG 23

- Fix: @daledah [$500] [UPDATED OFFER](https://www.upwork.com/nx/wm/offer/103636515)
- PR Review: @s77rt [$500] Payment via NewDot

**Comment by abekkala** [comment]
@daledah payment sent and contract ended - thank you! 🎉

**Comment by JmillsExpensify** [comment]
$500 approved for @s77rt

**Comment by tgolen** [comment]
This is still happening for me. Notice my LHN in focus mode:

![image](https://github.com/user-attachments/assets/7b677556-905b-4183-9dad-c629f205baf9)

While the chat search has many bolded reports (I see about 30 as I scroll down):
![image](https://github.com/user-attachments/assets/1e2ba9ae-3dd1-4b0c-8299-3342e1c5d0a8)

**Comment by s77rt** [comment]
@tgolen This is looking to be working as expected, chats that are bold/normal are consistent in the LHN and the find page.

**Comment by s77rt** [comment]
@tgolen Chats that are bold in screenshot1 are also bold in screenshot2 and vice-versa. If you are referring to the missing chats, that's a different bug.

**Comment by tgolen** [comment]
OK, I see what you're saying. You're right, I'm specifically speaking about unread chats missing from my LHN. I can open a new issue for that.

**Comment by s77rt** [comment]
@tgolen The chats not being listed is probably because their `notificationPreference` is set to `hidden`. Maybe that's the expected behaviour?

https://github.com/Expensify/App/blob/3b15d846bbe591631f05604fd6a512c2acbd6247/src/libs/SidebarUtils.ts#L140-L142

**Comment by tgolen** [comment]
You could be right about that. I spot checked a few, and it looks like it matches your theory. So, in this case, I would say this makes for a pretty confusing UX for the user. I wonder if maybe we shouldn't be showing them as bold if you have `hidden` preference. That would certainly have kept me from bringing this up at all.

Maybe this is something we should discuss back in Slack to get consensus on?

cc @ZhenjaHorbach and @nkdengineer

**Comment by s77rt** [comment]
Makes sense but let's ask in Slack to be sure
